pkg/operator/k8sutil/pod_test.go

```diff
@@ -80,3 +81,30 @@ func TestGetPodPhaseMap(t *testing.T) {
 	// list of failed pods should have 1 entry
 	assert.Equal(t, 1, len(podPhaseMap[v1.PodFailed]))
 }
+
+func TestAddUnreachableNodeToleration(t *testing.T) {
+	podSpec := v1.PodSpec{}
+
+	var tolerationSeconds int64 = 5
+	urToleration := v1.Toleration{Key: "node.kubernetes.io/unreachable",
+		Operator:          "Exists",
+		Effect:            "NoExecute",
+		TolerationSeconds: &tolerationSeconds}
+
+	// Test one toleration of 5 seconds
+	os.Setenv("ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE", "5s")
+	AddUnreachableNodeToleration(&podSpec)
+
+	assert.Equal(t, 1, len(podSpec.Tolerations))
+	assert.Equal(t, urToleration, podSpec.Tolerations[0])
+
+	// Test adding one additional toleration, replaces the previous one,
+	// keepin only the last.
+	os.Setenv("ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE", "6s")
+	tolerationSeconds = 6
```

Comment:
In reading this code, I find it confusing that this value is changed and is a pointer inside the struct. I think it would read more cleanly if you just modified the struct `urToleration` directly rather than relying on the pointer.

Reply:
Well, for me is also a weird thing why the "tolerationSeconds" is a pointer in the Toleration struct.
https://godoc.org/k8s.io/api/core/v1#Toleration

So..why do not take advantage of this weird thing to write less lines? :-D 
i have added a comment to make this clear ( use of the damn pointer )

Reply:
I agree that it being a pointer is strange. I'm not sure why they did it like this. However, fewer lines is not good reason to write unclear code. All it takes is dereferencing the pointer and setting it instead of changing the variable created for reference.

Reply:
Wait, since `tolerationSeconds` is **not** a pointer, updating this value should **not** update the pointer in the struct. Is this value even being used? Otherwise I wouldn't expect it to be having the desired effect. The thing that is changing the behavior is updating the env var, right? Looks like the test isn't using `tolerationSeconds` to change any behavior.

Reply:
I think this is more reason why this should be changed. It's obviously confusing to more than one of us.

Reply:
Seems like 2 of us being confused about what the code is actually doing is good reason to fix this to be more clear. Confusing code is much worse than creating more lines of code.

Reply:
Changed. I hope more clear in this way :-)